sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -746,6 +746,10 @@ export class EventHubReceiver extends LinkEntity {
             }
           },
           (err) => {
```

Comment:
I followed the code flow from here and it looks like we call the users error handler for any such error.
Which feels unnecessary if the error was retryable and we could just fetch events in the next `receiveBatch()` call. Am I missing something here?

Reply:
Or is the expectation that we report every error we find when receiving events, but keep the loop going and call users close handler only if the error is not retryable?

Reply:
We report every error we see after exhausting retries. We decided to do this so that users would have some way to tell that there’s an issue going on rather than have the appearance that the receiver silently stopped receiving messages. This way they can log or publish a metric and see that there was an issue preventing new events from coming in.

Reply:
ok